src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -1168,5 +1162,156 @@ public void NameofTypeParameterInParameterInitializer()
 }";
             var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType1()
+        {
+            var source =
+@"class A<T> { }
+class Test {
+  void M(string s = nameof(A<>)) {
+  }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType2()
+        {
+            var source =
+@"class A<T> { }
+class Test {
+  void M(string v = nameof(A<,>)) {
+  }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics(
+                // (3,28): error CS0305: Using the generic type 'A<T>' requires 1 type arguments
+                //   void M(string v = nameof(A<,>)) {
+                Diagnostic(ErrorCode.ERR_BadArity, "A<,>").WithArguments("A<T>", "type", "1").WithLocation(3, 28));
+        }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType3()
+        {
+            var source =
+@"class B<X,Y> { }
+class Test {
+  void M(string s = nameof(B<>)) {
+  }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics(
+                // (3,28): error CS0305: Using the generic type 'B<X, Y>' requires 2 type arguments
+                //   void M(string s = nameof(B<>)) {
+                Diagnostic(ErrorCode.ERR_BadArity, "B<>").WithArguments("B<X, Y>", "type", "2").WithLocation(3, 28));
+        }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType4()
+        {
+            var source =
+@"class B<X,Y> { }
+class Test {
+  void M(string v = nameof(B<,>)) {
+  }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType5()
+        {
+            var source =
+@"class C<T> { public class Nested { } }
+class Test {
+  void M(string s = nameof(C.Nested)) {
+  }
+}";
+            var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics(
+                // (3,28): error CS0305: Using the generic type 'C<T>' requires 1 type arguments
+                //   void M(string s = nameof(C.Nested)) {
+                Diagnostic(ErrorCode.ERR_BadArity, "C").WithArguments("C<T>", "type", "1").WithLocation(3, 28));
+        }
+
+        [Fact, WorkItem(4827, "https://github.com/dotnet/roslyn/issues/4827")]
+        public void NameofOpenGenericType6()
+        {
+            var source =
+@"class C<T> { public class Nested { } }
+class Test {
+  void M(string s = nameof(C<>.Nested)) {
```

Comment:
What does SemanticModel returns for the argument of nameof and its subexpressions